actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -11,6 +11,7 @@ class UnsafeRedirectError < StandardError; end
 
     included do
       mattr_accessor :raise_on_open_redirects, default: false
+      mattr_accessor :redirect_code_for_unsafe_http_methods, default: 302
```

Comment:
`mattr_accessor` is rarely the right choice.

I think here might want to use `class_attribute`, so that we can set the default on `AC::Base`, and then refine on a per controller basis if needed.

Reply:
> For clarity, would you say that the order of preference is

1. `singleton_class.attr_accessor` - when it's exactly one class, or more frequently, when it's single global.
2. `class_attribute` - when the scope is a class hierarchy, allowing value inheritance, and it's not meant to be changed too much at runtime (`class_attributes` writes are slow because it's basically a `define_method`).
3. `mattr_accessor` - honestly there's no reason to use it ever.

> do you feel like `:raise_on_open_redirects`

It could make sense yes. As you mentioned it could allow to migrate on a per controller basis.